### PR TITLE
Independent project build process for example generator IT

### PR DESCRIPTION
**Pull request by totalo**
For #12366.

Changes proposed in this pull request:
  - 
  -
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `mvn clean install -B -T2C -DskipTests -Dmaven.javadoc.skip=true -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: .github/workflows/nightly-it-example-generator.yml [hunk 1]
```diff
@@ -25,11 +25,32 @@ env:
...
+      - uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: shardingsphere-maven-third-party-cache-${{ github.sha }}
+          restore-keys: |
+            shardingsphere-maven-third-party-cache-
+            shardingsphere-maven-third-party-
+      - name: Build with Maven
+        run: ./mvnw -B -T1C -ntp clean install -DskipITs -DskipTests
   example-generator:
```
Comment (terrymanu) [inline .github/workflows/nightly-it-example-generator.yml:48]: how about rename job name as `generate-and-test-examples`

Comment (totalo) [reply .github/workflows/nightly-it-example-generator.yml]: done.

File: .github/workflows/nightly-it-example-generator.yml [hunk 2]
```diff
@@ -25,11 +25,32 @@ env:
...
+      - uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: shardingsphere-maven-third-party-cache-${{ github.sha }}
+          restore-keys: |
+            shardingsphere-maven-third-party-cache-
+            shardingsphere-maven-third-party-
+      - name: Build with Maven
+        run: ./mvnw -B -T1C -ntp clean install -DskipITs -DskipTests
+  Generate-and-test-examples:
```
Comment (terrymanu) [inline .github/workflows/nightly-it-example-generator.yml:48]: Please use lower case for all job name

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: .github/workflows/nightly-it-example-generator.yml (hunk 2 above)
Comment (totalo) [reply .github/workflows/nightly-it-example-generator.yml]: done.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**